### PR TITLE
Mccalluc/transaction log

### PR DESCRIPTION
**Pull request by mccalluc**
This is a proof of concept about making the history of proxied requests available to the outside world. For now ...
- they are just going in a file, when we'd probably want some kind of DB in the long run
- and to get it back out right now there is only `list` which just `readlines` the file, when we probably want to query the database, and return result records.

### CODE REVIEW

**Comment by mccalluc** [comment]
*Please* hold off on this until #56 and #57 are resolved: They are refactorings across multiple files, and it should be pretty easy to drop this on top once those are in, but the reverse could be trickier.

File: django_docker_engine/proxy.py [hunk 1]
```diff
@@ -0,0 +1,50 @@
...
+
+    def log(self, *args):
+        with open(self.path, 'a') as f:
+            timestamp = datetime.now().isoformat()
+            args_list = list(args)
+            args_list.insert(0, timestamp)
+            print('\t'.join(args_list), file=f)
+
+    def list(self):
+        return open(self.path).readlines()
```
Comment (scottx611x) [inline django_docker_engine/proxy.py]: does this need to be `close()` or use `with`

File: tests/test_docker_utils.py [hunk 2]
```diff
@@ -91,8 +92,8 @@ def count_my_containers(self):
             filters={'label': self.test_label}
         ))
 
-    def one_file_server(self, container_name, html):
-        self.write_to_host(html, os.path.join(self.tmp, 'index.html'))
+    def one_file_server(self, container_name, html, filename='index.html'):
```
Comment (scottx611x) [inline tests/test_docker_utils.py]: Has been removed in master

File: tests/test_docker_utils.py [hunk 3]
```diff
@@ -152,9 +153,13 @@ def test_container_spec_with_input(self):
     def test_docker_proxy(self):
         container_name = self.timestamp()
         hello_html = '<html><body>hello proxy</body></html>'
-        self.one_file_server(container_name, hello_html)
-        url = '/docker/{}/'.format(container_name)
+        filename = 'hello.html'
```
Comment (scottx611x) [inline tests/test_docker_utils.py]: Has been removed in master

**Pull request merged**